airbyte-server/src/main/java/io/airbyte/server/handlers/WorkspacesHandler.java

```diff
@@ -249,4 +252,32 @@ private static WorkspaceRead buildWorkspaceRead(final StandardWorkspace workspac
         .notifications(NotificationConverter.toApiList(workspace.getNotifications()));
   }
 
+  private void validateWorkspacePatch(final StandardWorkspace persistedWorkspace, final WorkspaceUpdate workspacePatch) {
+    Preconditions.checkArgument(persistedWorkspace.getWorkspaceId().equals(workspacePatch.getWorkspaceId()));
+  }
+
+  private void applyPatchToStandardWorkspace(final StandardWorkspace workspace, final WorkspaceUpdate workspacePatch) {
+    if (workspacePatch.getAnonymousDataCollection() != null) {
+      workspace.setAnonymousDataCollection(workspacePatch.getAnonymousDataCollection());
+    }
+    if (workspacePatch.getNews() != null) {
+      workspace.setNews(workspacePatch.getNews());
+    }
+    if (workspacePatch.getDisplaySetupWizard() != null) {
+      workspace.setDisplaySetupWizard(workspacePatch.getDisplaySetupWizard());
+    }
+    if (workspacePatch.getSecurityUpdates() != null) {
+      workspace.setSecurityUpdates(workspacePatch.getSecurityUpdates());
+    }
+    if (workspacePatch.getEmail() != null) {
```

Comment:
It looks like the old code was doing a `nullOrEmpty` check, but the new code is just checking for null. Maybe the best thing to do for the new code would be to add a validation in `validateWorkspacePatch` that the email address is properly formed? Or at least not an empty string?